### DISCUSSION TITLE
question abount key serialization

### DISCUSSION THREAD

**Discussion by anc95** [vercel/swr#598] [General] [answered]
when I use swr as follow, it will revalidate every time because of swr's key serialization mechanism
```js
function App() {
  const {data} = useSWR(['/api/list', {id: '123'}]);
  // ...
}
```

1. I konw that react-query and apollo both use Deep Compare (JSON.stringify(key)) to serialize key, so do u have plan to use thiss on swr?
2. or can u provide a custom serialize api `queryKeySerializerFn` ？
3. or is there any other solution to fix my issue?

**Comment by pke** [comment, marked as answer]
You can also just serialise your key.

`const {data} = useSWR(['/api/list', JSON.stringify({id: '123'})]);`

You will then get the stringified object as second string argument to your fetch function like this:

```ts
const {data} = useSWR(
  ['/api/list', JSON.stringify({id: '123'})],
  (path, params) => {
    fetch(path, { body: JSON.parse(params) })
  }
)
```

You just convert it back to an object. Which is guaranteed to not throw, since you just converted it.
